src/Articles.jsx

```diff
@@ -1,25 +1,46 @@
 import { getArticles } from './Api'
 import { useEffect, useState } from "react"
+import { Link } from 'react-router-dom';
+
 
 function Articles() {
-const [articles, setArticles] = useState([])
+const [articles, setArticles] = useState([]);
+const [isLoading, setIsLoading] = useState(true);
+const [error, setError] = useState(null)
 
 useEffect(() => {
     getArticles().then((articlesFromApi) => {
-    setArticles(articlesFromApi)
+    setArticles(articlesFromApi);
+    setIsLoading(false);
+    })
+    .catch((err) => {
+        setError(err);
+        setIsLoading(false)
     })
 
 }, [])
 
+if (isLoading) {
+    return <div>Won't be long, just loading...</div>
+}
+
+if (error)
+return <div>Error: {error.message} </div>
+
```

Comment:
You could add this _error state_ to your single article page too! Sometimes a user might be cheeky and try to access an ID in the URL that doesn't exist, causing the request to fail 💡